crates/wasmtime/src/runtime/module.rs

```diff
@@ -1027,6 +1027,25 @@ impl Module {
         })
     }
 
+    /// Get the locations of functions in this module's `.text` section, with the
+    /// function names.
+    ///
+    /// Each function's location is a (name, `.text` section offset, length) pair.
+    pub fn function_locations_with_names<'a>(
+        &'a self,
+    ) -> impl ExactSizeIterator<Item = (String, usize, usize)> + 'a {
+        let module = self.compiled_module();
+        self.function_locations().map(|(offset, len)| {
+            let (idx, _) = module.func_by_text_offset(offset).unwrap();
```

Comment:
We can avoid looking up the function index via instruction offset since `function_locations` should already return the functions in order:

```suggestion
        self.function_locations().enumerate().map(|(idx, (offset, len))| {
            let idx = u32::try_from(idx).unwrap();
```